src/Infrastructure.VS/LanguageDetection/SonarLanguageRecognizer.cs

```diff
@@ -124,12 +124,13 @@ private static bool IsTypeScriptDocument(IEnumerable<IContentType> contentTypes)
         {
             var extension = GetNormalizedExtention(fileName);
 
+            var contentTypeName = fileExtensionRegistryService.GetContentTypeForExtension(extension).TypeName;
+
             // ContentType for "js" is typescript we do manual check to be consistent with Detect method
             if (JavascriptSupportedExtensions.Contains(extension)) { return AnalysisLanguage.Javascript; }
-            if (fileExtensionRegistryService.GetContentTypeForExtension(extension).TypeName == TypeScriptTypeName) { return AnalysisLanguage.TypeScript; }
-            if (fileExtensionRegistryService.GetContentTypeForExtension(extension).TypeName == CFamilyTypeName) { return AnalysisLanguage.CFamily; }
-            if (fileExtensionRegistryService.GetContentTypeForExtension(extension).TypeName == CSharpTypeName 
-                || fileExtensionRegistryService.GetContentTypeForExtension(extension).TypeName == BasicTypeName) { return AnalysisLanguage.RoslynFamily; }
+            if (contentTypeName == TypeScriptTypeName) { return AnalysisLanguage.TypeScript; }
```

Comment:
I'd move the call to GetContentTypeForExtension after the javascript call, just for potential performance. Also a switch case might make this more readable, with default returning null.